src/components/FunctionSeries.tsx

```diff
@@ -1,19 +1,32 @@
+import { useMemo } from 'react';
+
+import { usePlotContext } from '../contexts/plotContext';
 import { SeriesPoint } from '../types';
 
 import { LineSeries, LineSeriesProps } from './LineSeries';
 
 export interface FunctionSeriesProps extends Omit<LineSeriesProps, 'data'> {
   data: (x: number) => number;
-  max: number;
-  min: number;
   step?: number;
 }
 
 export function FunctionSeries(props: FunctionSeriesProps) {
-  const { data: oldData, max, min, step = 0.1, ...otherProps } = props;
-  const data: SeriesPoint[] = [];
-  for (let i = min; i < max; i += step) {
-    data.push({ x: i, y: oldData(i) });
+  const { data: oldData, step = 0.1, ...otherProps } = props;
+  const {
+    axisContext: { x },
+    plotWidth,
+  } = usePlotContext();
+  const data = useMemo(() => {
+    const min = x ? x.scale.invert(0) : 0;
+    const max = x ? x.scale.invert(plotWidth) : 1;
```

Comment:
when we don't have min/max in x axis the min=0 ,max=1
i tired to do what @lpatiny ask ( max=min=0) but in this case y max and min wont generate automaticly so we will need to set max/min of y and x axis (illogic)

Reply:
the result when x axis max/min are undefined 
![image](https://user-images.githubusercontent.com/56769876/151829333-405cd9d3-3a98-47ae-87e5-bdd5d45a7052.png)